src/Validation/Constraint/HasClaimWithValue.php

```diff
@@ -0,0 +1,30 @@
+<?php
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Validation\Constraint;
+
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\Validation\Constraint;
+use Lcobucci\JWT\Validation\ConstraintViolation;
+
+final class HasClaimWithValue implements Constraint
+{
+    private string $claimId;
+
+    private string $claimValue;
+
+    public function __construct(string $claimId, string $claimValue)
+    {
+        $this->claimId    = $claimId;
+        $this->claimValue = $claimValue;
+    }
+
+    public function assert(Token $token): void
+    {
+        if (! $token->hasClaimWithValue($this->claimId, $this->claimValue)) {
```

Comment:
I really like the idea. Just believe that the check could be contained here - which would reduce the changes quite drastically.

We should also verify that we receive a unencrypted token (similar to what's done on the SignedWith constraint). That's important because we don't have access to claims on encrypted tokens.